hollow-diff-ui/src/main/java/com/netflix/hollow/history/ui/HollowHistoryUI.java

```diff
@@ -76,15 +83,21 @@ public HollowHistoryUI(String baseUrlPath, HollowConsumer consumer, int numState
         this(baseUrlPath, createHistory(consumer, numStatesToTrack), timeZone);
     }
     
-    private static HollowHistory createHistory(HollowConsumer consumer, int numStatesToTrack) {
-        consumer.getRefreshLock().lock();
-        try {
-            HollowHistory history = new HollowHistory(consumer.getStateEngine(), consumer.getCurrentVersionId(), numStatesToTrack);
-            consumer.addRefreshListener(new HollowHistoryRefreshListener(history));
-            return history;
-        } finally {
-            consumer.getRefreshLock().unlock();
-        }
+    /**
+     * HollowHistoryUI that supports building history in both directions simultaneously.
+     * Fwd and rev consumers should be initialized to the same version before calling this constructor.
+     * This constructor defaults max states to 1024 and time zone to PST.
+     *
+     * @param baseUrlPath url path for history UI endpoint
+     * @param consumerFwd HollowConsumer (already initialized with data) that will be traversing forward deltas or double snapshots
+     * @param consumerRev HollowConsumer (also initialized to the same version as consumerFwd) that will be traversing reverse deltas
+     */
+    public HollowHistoryUI(String baseUrlPath, HollowConsumer consumerFwd, HollowConsumer consumerRev) {
+        this(baseUrlPath, consumerFwd, consumerRev, 1024, VersionTimestampConverter.PACIFIC_TIMEZONE);
```

Comment:
nit : could we make the max states configurable?

Reply:
Yup they are, this constructor is just plugging in a default to the other constructor that accepts it as a param

Reply:
Hah, we were missing the max states configurable flavor constructor in the `HollowHistoryUIServer` class. Not sure if you may have been thinking of here. Now added- https://github.com/Netflix/hollow/commit/d4d3dbed944a8885ba128843a61e569ddbcce08e